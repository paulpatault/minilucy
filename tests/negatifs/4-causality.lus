node main0 () returns (a : int; b : int);
let
  a = b;
  b = a;
tel;
