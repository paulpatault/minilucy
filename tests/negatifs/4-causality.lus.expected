
/**************************************/
/* Sugar automaton print              */
/**************************************/
type inductive_bool =
  | False
  | True

node main0() returns (a : int; b : int)
var ;
let
  (a) = b;
  (b) = a;
tel

/**************************************/
/* Typed ast                          */
/**************************************/
type inductive_bool =
  | False
  | True

node main0__7() returns (a__5 : int; b__6 : int)
var ;
let
  (a__5) = b__6;
  (b__6) = a__5;
tel

/**************************************/
/* Normalized ast                     */
/**************************************/
type inductive_bool =
  | False
  | True

node main0__7() returns (a__5 : int; b__6 : int)
var ;
let
  (a__5) = b__6;
  (b__6) = a__5;
tel

(a__5) = b__6(b__6) = a__5